src/poetry/mixology/incompatibility.py

```diff
@@ -385,7 +380,11 @@ def _try_requires_through(
         return "".join(buffer)
 
     def _try_requires_forbidden(
-        self, other: Incompatibility, details: dict, this_line: int, other_line: int
+        self,
+        other: Incompatibility,
+        details: dict,
```

Comment:
Can we get more specific here?

Reply:
hi @abn , looked at in details, specific in terms of types is not possible here, the arguments take up the type of the arguments passed by the calling functions, tried to match the types from there 
src/poetry/mixology/incompatibility.py <line number 216>
   def and_to_string(
        self,
        other: Incompatibility,
        details: dict,
        this_line: int | None,
        other_line: int | None,
    ) -> str:

and details:dict is not even used its redundant

Reply:
since `dict` was already here, I'm fine with any improvements as a follow up